src/main/java/com/slack/kaldb/metadata/core/CorruptMetadataNodeException.java

```diff
@@ -0,0 +1,11 @@
+package com.slack.kaldb.metadata.core;
+
+public class CorruptMetadataNodeException extends RuntimeException {
```

Comment:
We probably don't need a custom exception here - per Effective Java we should favor standard exceptions (#72)  as it makes the API easier to learn, has a smaller memory footprint, and reduces the time loading. An `IllegalStateException` likely makes the most sense here.

Reply:
I created a new exception in this case to indicate a distinct application failure state. In this case, this exception is used to perform cache invalidation in CacheableMetdataStore if the data is malformed.

Reply:
Wouldn't an `InvalidStateException` do the same thing? We're never actually catch a `CorruptMetadataNodeException` - we just throw it and then catch the `RuntimeException`. 

 As a consumer of this class any `RuntimeException` would necessitate the invalidation of the metadata store.

Reply:
Good point. Replaced it with an IllegalStateException. I added it initially to do cache invalidation, but ended up not using it.